test/specs/parsing/functions/functions.string.tests.ts

```diff
@@ -509,4 +509,15 @@ describe('functions over strings', () => {
 		it('Returns false() for the expression fn:codepoint-equal("a", "b")', () =>
 			chai.assert.isFalse(evaluateXPathToBoolean('codepoint-equal("a","b")', documentNode)));
 	});
+
+	describe('encode-for-uri()', () => {
+		it('Returns the correct example for the example in the specs', () => {
+			chai.assert.equal(evaluateXPathToString('encode-for-uri("http://www.example.com/00/Weather/CA/Los%20Angeles#ocean")'), "http%3A%2F%2Fwww.example.com%2F00%2FWeather%2FCA%2FLos%2520Angeles%23ocean")
+
+		})
+		
+		it('Returns empty for an empty sequence', () => {
```

Comment:
If $uri-part is the empty sequence, the function returns the zero-length string.
fn:empty("") returns false().